### DISCUSSION TITLE
Source connection as environment variable

### DISCUSSION THREAD

**Discussion by fernandocalsa** [Budibase/budibase#6556] [Ideas]
When we connect to a new source (Mongo, Postgres...) we need to provide the connection parameters (host, port, user, password...). This can be done straight forward from the UI, but when these credentials are secret, we need to find another way to fill these params.

When we self-host Budibase we could have an environment variable with the credentials, and when creating the new source we could type a binding variable to that env variable

![Screenshot 2022-07-05 at 11 06 40](https://user-images.githubusercontent.com/6080414/177292699-45b3152a-eb72-4958-a62c-b1ef2ef930a1.png)

Do you think this is feasible?

**Comment by melohagan** [comment]
Great idea! I think this issue is related as well: https://github.com/Budibase/budibase/issues/4188

**Reply by fernandocalsa** [reply]
Yes! but I think this issue should be simpler, as I think we don't need any secret manager in Budibase, environment variables is the way we usually store this kind of secrets, so from Docker or k8s we could access to them.
